fvm/src/kernel/default.rs

```diff
@@ -342,8 +355,12 @@ where
         Ok(k)
     }
 
-    fn block_read(&self, id: BlockId, offset: u32, buf: &mut [u8]) -> Result<u32> {
+    fn block_read(&mut self, id: BlockId, offset: u32, buf: &mut [u8]) -> Result<u32> {
         let data = self.blocks.get(id).or_illegal_argument()?.data();
+
+        self.call_manager
+            .charge_gas(self.call_manager.price_list().on_block_read(data.len()))?;
```

Comment:
We've already charged to "load" the block, now we just want to charge for the amount of data read (the length we return from this function).

Reply:
AH, no. I mean we need to charge for the amount of data read which might, technically, be slightly different from the size of the block.